### PR TITLE
Laravel 9.x Shift

### PR DESCRIPTION
**Pull request by georgeburdell**
This pull request includes the changes for upgrading to Laravel 9.x. Feel free to commit any additional changes to the `shift-63224` branch.

**Before merging**, you need to:

- Checkout the `shift-63224` branch
- Review **all** pull request comments for additional changes
- Run `composer update` (if the scripts fail, add `--no-scripts`)
- Clear any config, route, or view cache
- Thoroughly test your application ([no tests?](https://laravelshift.com/laravel-test-generator), [no CI?](https://laravelshift.com/ci-generator))

If you need help with your upgrade, check out the [Human Shifts](https://laravelshift.com/human-shifts). You may also join the [Shifty Coders](https://laravelshift.com/shifty-coders) Slack workspace to level-up your Laravel skills.

### CODE REVIEW

**Comment by georgeburdell** [comment]
:information_source: Laravel 9 moved the `resources/lang` folder to the top level of the project. While Shift moved and replaced references to this folder, you may have additional references to this folder which need to be updated.

**Comment by georgeburdell** [comment]
:warning: Laravel 9 upgraded to Flysystem 3.0 causing the behavior of the `Storage` facade to change slightly:

* Write operations (such as `put` and `write`) now overwrite existing files by default, and now return `false` instead of an exception on failure.
* Read operations (such as `get` and `download`) now return `null` instead of a `FileNotFoundException` when a file is missing.
* The `delete` operation now returns `true` even if the file did not exist.

For more details, you may review the [Flysystem 3.x](https://laravel.com/docs/9.x/upgrade#flysystem-3) section of the _Upgrade Guide_.

**Comment by georgeburdell** [comment]
:warning: Laravel 9 transitioned from SwiftMailer to Symfony Mailer. While this transition should be relatively seamless and Shift automates any tedious changes, there are some changes which are not reliably automated.

**Dependencies**
Some mail providers may require an additional dependency. For example, if you are using Mailgun, then you will need to install the `symfony/mailgun-mailer` package:

```sh
composer require symfony/mailgun-mailer symfony/http-client
```

**Updated Return Types**
The `send`, `html`, `text`, and `plain` methods no longer return the number of recipients that received the message.

**Renamed "Swift" Methods and Classes**
Various SwiftMailer related methods and classes have been renamed to their Symfony Mailer counterparts. Shift renamed these for you, but depending on your usage, you may need to update your code for some changes in the APIs.

**Failed Recipients**
It is no longer possible to retrieve a list of failed recipients after sending a message. Instead, a `Symfony\Component\Mailer\Exception\TransportExceptionInterface` exception will be thrown if a message fails to send. Instead of relying on retrieving invalid email addresses after sending a message, it is recommend that you validate email addresses before sending the message instead.

For more details on what has changed you may review the [Symfony Mailer](https://laravel.com/docs/9.x/upgrade#symfony-mailer) section of the _Upgrade Guide_.

**Comment by georgeburdell** [comment]
:x: Shift could not upgrade the following files since they differed from the default Laravel version. You will need to compare these files against the [default Laravel 9 versions](https://github.com/laravel-shift/laravel-9.x/) and merge any changes:

- [ ] lang/en/auth.php
- [ ] lang/en/validation.php
- [ ] public/index.php

**Comment by georgeburdell** [comment]
:warning: The `FILESYSTEM_DRIVER` environment variable was renamed to `FILESYSTEM_DISK` in Laravel 9. Shift attempted to [automate this change](https://github.com/RoboJackets/apiary/pull/2527/commits/960c2a949d5d816cf691cb9c5e7f570e661ba2ab?w=1), but you should review any additional environment configuration and rename this variable.

**Comment by georgeburdell** [comment]
:warning: Shift upgraded your configuration files by defaulting them and merging your _true customizations_. These include values which are not changeable through core `ENV` variables.

You should [review this commit](https://github.com/RoboJackets/apiary/pull/2527/commits/a4ac0ed73e3e7df60ba496af8458976fc9998bc9?w=1) for additional customizations or opportunities to use new `ENV` variables. If you have a lot of customizations, you may undo this commit with `git revert a4ac0ed7` and make the [config file changes](https://github.com/laravel-shift/laravel-7.x/pull/2/commits/b6d066d9fa2b9482224f7b7332a3fa287e374924) manually.

**Comment by georgeburdell** [comment]
:information_source: The `FILESYSTEM_CLOUD` environment variable was removed in a later release of Laravel 8. While you may still use the `cloud` disk, you are encouraged to [configure your own disks](https://laravel.com/docs/9.x/filesystem#configuration).

Shift found references to the `FILESYSTEM_CLOUD` variable in the following files:

- [ ] ./config/filesystems.php

**Comment by georgeburdell** [comment]
:warning: The `token` driver for API authentication is no longer available. This driver was not very robust and was removed from the documentation. Laravel now recommends using [Sanctum](https://laravel.com/docs/9.x/sanctum).

If you were using the `token` driver, you may re-add it within your `auth.php` configuration file then migrate to _Sanctum_ at your convenience.

**Comment by georgeburdell** [comment]
:information_source: Shift [updated your dependencies](https://github.com/RoboJackets/apiary/pull/2527/commits/abd457ed91b1ffbe91b68ed66c8cd48c0d1a57dd?w=1) for Laravel 9. While many of the popular packages are reviewed, you may have to update additional packages in order for your application to be compatible with Laravel 9.

Watch [dealing with dependencies](https://laravelshift.com/videos/update-incompatible-composer-dependencies) for tips on handling any additional package incompatibilities.

**Comment by georgeburdell** [comment]
:warning: Shift detected you are using a Laravel package like Horizon or Nova which may need to have its published assets regenerated after upgrading. Be sure to use `artisan` to _republish_ these assets as well as `php artisan view:clear` to avoid any errors.

**Comment by georgeburdell** [comment]
:tada: Congratulations, you're now running the latest version of Laravel!

Next, you may optionally run the following _Shifts_ to ensure your application is fully upgraded, adopts the latest Laravel conventions, and easier to maintain in the future:

- [Upgrade Checker](https://laravelshift.com/laravel-upgrade-checker) ensures your application is fully upgraded by detecting any outdated code.
- [Laravel Fixer](https://laravelshift.com/laravel-code-fixer) automatically updates your code to the latest Laravel conventions.
- [Tests Generator](https://laravelshift.com/laravel-test-generator) intelligently generates model factories, HTTP Tests, and configuration for your application.
- [CI Generator](https://laravelshift.com/ci-generator) intelligently generates CI jobs to lint PHP, check code style, and run tests, including Dusk.

You may also use the [Shift Workbench](https://laravelshift.com/workbench) to automate common tasks for maintaining your Laravel and PHP applications.

File: app/Nova/User.php [hunk 1]
```diff
@@ -385,16 +384,6 @@ protected function metaFields(): array
 
             Text::make('gtDirGUID')
                 ->hideFromIndex(),
-
-            MorphToMany::make('Roles', 'roles', \Vyuldashev\NovaPermission\Role::class)
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Nova/User.php:389]: This package does not work with both Laravel 9 and Nova 3, so I have to update Nova to use it. I will do that as part of #2464.

File: composer.json [hunk 2]
```diff
@@ -46,36 +46,38 @@
...
-        "square/square": "17.3.0.20220316",
+        "spatie/laravel-failed-job-monitor": "^4.0",
+        "spatie/laravel-permission": "^5.5",
+        "spatie/laravel-webhook-client": "^3.1",
+        "spatie/once": "^3.1",
+        "square/square": "^19.0",
         "subfission/cas": "4.2.0",
-        "ukfast/laravel-health-check": "1.13.3",
-        "uxweb/sweet-alert": "2.0.5",
-        "vyuldashev/nova-permission": "2.11.1"
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply composer.json:61]: This package does not work with both Laravel 9 and Nova 3, so I have to update Nova to use it. I will do that as part of #2464.

File: app/Nova/User.php [hunk 3]
```diff
@@ -385,16 +384,6 @@ protected function metaFields(): array
...
-                ->canSee(static function (Request $request): bool {
-                    return $request->user()->hasRole('admin');
-                }),
-
-            MorphToMany::make('Permissions', 'permissions', \Vyuldashev\NovaPermission\Permission::class)
-                ->canSee(static function (Request $request): bool {
-                    return $request->user()->hasRole('admin');
-                }),
```
Comment (kberzinch) [inline app/Nova/User.php:397]: Put this back when upgrading Nova

File: app/Providers/NovaServiceProvider.php [hunk 4]
```diff
@@ -175,9 +175,6 @@ static function (Request $request): bool {
     public function tools(): array
     {
         return [
-            (new NovaPermissionTool())->canSee(static function (Request $request): bool {
-                return $request->user()->hasRole('admin');
-            }),
```
Comment (kberzinch) [inline app/Providers/NovaServiceProvider.php:180]: Put this back when upgrading Nova

**Review by kberzinch — APPROVED** [review]

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**